### PR TITLE
Ataymano/policy cleanup

### PR DESCRIPTION
**Pull request by ataymano**
no vw stuff in RLLoop

### CODE REVIEW

File: src/learn_to_pick/pick_best.py [hunk 1]
```diff
@@ -379,6 +346,65 @@ def create(
...
+            policy = policy,
             selection_scorer=selection_scorer,
             **kwargs,
         )
+    
+    @staticmethod
+    def create_policy(
+        feature_embedder: Optional[base.Embedder] = None,
+        vw_cmd: Optional[List[str]] = None,
+        model_save_dir: str = "./",
```
Comment (olgavrou) [inline src/learn_to_pick/pick_best.py:358]: generic things like `model_save_dir`, `reset_model`, `logs_dir` (doesn't have to be VW) can be set in `PickBest` args and be passed down to `create_policy`, could be achieved by just passing `kwargs` from `PickBest` into `create_policy` (which we probably should do anyway)

Comment (olgavrou) [reply src/learn_to_pick/pick_best.py:358]: That way the user doesn't have to specify a policy object to specify these things which are module specific and not policy specific necessarily, even though they are driven by the policy

Comment (ataymano) [reply src/learn_to_pick/pick_best.py:358]: We can do either way but should be consistent about it:
1) We can either have saving model / logging etc outside of the policy. It is not aligned with existing implementation (where all of this is already members of VwPolicy and member of RLLoop at the same time). It is also not clear to me how it can be done completely agnostic from the policy since it is tightly coupled with model implementation and most likely implementation still will be inside policy (even if there is interface outside). Also trivial baseline policies (like random one) are not having any model which suggests it to be policy level implementation detail.
2) We can make policy only responsible for decision making but not for model-related infrastructure, but in this case:
- it should be removed from VwPolicy
- have less clarity how to do it in completely independent way (I am kind of repeating p.1 here)

Basically, right now we have all this stuff sitting in both RLLoop and VwPolicy which seems wrong.

Comment (olgavrou) [reply src/learn_to_pick/pick_best.py:358]: I agree for these to belong to the policy and have the policy deal with them, the comment was for them to be able to be set in the PickBest level without *having* to specify the policy

user can ignore them in the random policy scenario, but it would be nice to be able to do:

`PickBest.create(model_save_dir=<path>)` instead of having to do `PickBest.create(policy=VwPolicy(model_save_dir=<path>))`

Comment (ataymano) [reply src/learn_to_pick/pick_best.py:358]: oh yeah sure,  can do that
